src/gui/application.cpp

```diff
@@ -652,24 +564,26 @@ void Application::setupTranslations()
     }
 }
 
-bool Application::giveHelp()
+void Application::showMainDialog()
 {
-    return _helpOnly;
+    _gui->slotOpenMainDialog();
 }
 
-bool Application::versionOnly()
+void Application::slotGuiIsShowingSettings()
 {
-    return _versionOnly;
+    emit isShowingSettingsDialog();
 }
 
-void Application::showMainDialog()
+void Application::setupConfigDirectory()
 {
-    _gui->slotOpenMainDialog();
+    if (!_parser.value("confdir").isEmpty() && !ConfigFile::setConfDir(_parser.value("confdir"))) {
+        showHint("Invalid path passed to --confdir");
+    }
 }
 
-void Application::slotGuiIsShowingSettings()
+bool Application::getLogDebug()
 {
-    emit isShowingSettingsDialog();
+    return (_parser.isSet("logdebug") || _parser.isSet("debug") || ConfigFile().logDebug());
```

Comment:
I think I'd keep that inlined in the caller and not introduce a `getLogDebug()` function for that.

Note I got a hidden reason for this... the debug flag is in fact unused. No one uses `debugMode()` so a second PR could easily remove it. ;-)

Reply:
Okay, I will change this to be inline tomorrow.

Reply:
Hey, your hidden reason is good ;). I just made a refactoring of methods and did not question the defined methods. `debugMode()` is really not used anywhere the only functionality of it was to overwrite --logdebug if not specified. Do you really want to open a second PR or should I just remove it as well?

Reply:
Yep you can put the `debugMode()` cleanup in that PR if you want.